fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/PruneSubfieldsForComplexType.java

```diff
@@ -153,7 +170,7 @@ public Void visit(OptExpression optExpression, PruneComplexTypeUtil.Context cont
         public Void visitPhysicalScan(OptExpression optExpression, PruneComplexTypeUtil.Context context) {
             PhysicalScanOperator physicalScanOperator = (PhysicalScanOperator) optExpression.getOp();
 
-            if (OperatorType.PHYSICAL_OLAP_SCAN.equals(physicalScanOperator.getOpType())) {
+            if (OperatorType.PHYSICAL_OLAP_SCAN.equals(physicalScanOperator.getOpType()) && !FeConstants.runningUnitTest) {
```

Comment:
You can refer to this pr: https://github.com/StarRocks/starrocks/pull/31345

We can mock a hive table which contains struct columns, we don't need to use `FeConstants.runningUnitTest` to control logic.